packages/vxrn/src/utils/scanDepsToPreBundleForSsr.ts

```diff
@@ -123,7 +122,51 @@ export async function scanDepsToPreBundleForSsr(
           dep.startsWith('@expo/') ||
           dep.startsWith('expo-')
 
-        const result = [...(shouldPreBundle ? [dep] : []), ...subDepsToPreBundle]
+        const result = [
+          ...(shouldPreBundle
+            ? (() => {
+                // Not working, and do we really need this?
+                // const definedExports = Object.keys(depPkgJson.exports || {})
+                //   .map((k) => k.replace(/^\.\/?/, ''))
+                //   .filter((k) => !!k && k !== 'package.json')
+                //   .map((k) => `${dep}/${k}`)
+                const definedExports = []
+
+                /**
+                 * A dirty workaround for packages that are using entry points that are not explicitly defined,
+                 * such as while using react-native-vector-icons, users will import Icon components like this: `import Icon from 'react-native-vector-icons/FontAwesome'`.
+                 */
+                const specialExports = (() => {
```

Comment:
hmm... do they not define this in package.json exports? couldn't we find it there?

Reply:
I *guess* if they defined it, Vite might be smart enough to also optimize those defined entry points.